### PR TITLE
Added java coding program 9

### PR DESCRIPTION
**Pull request by tejashrighogare**

### CODE REVIEW

File: checkPalindrome/CheckPalindromeProgram.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public static void isPalindrome(int inputNumber) {
+        int palindromeNumber = inputNumber;
+        int reverseNumber = 0;
+        while (palindromeNumber != 0) {
+            int remainder = palindromeNumber % 10;
+            reverseNumber = reverseNumber * 10 + remainder;
+            palindromeNumber = palindromeNumber / 10;
+        }
```
Comment (prashantkawa) [inline checkPalindrome/CheckPalindromeProgram.java]: Try with different logic

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: checkPalindrome/CheckPalindromeProgram.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        CheckPalindromeProgram palindromeObject = new CheckPalindromeProgram();
+        temporaryNumber = inputNumber;
+        finalNumber = palindromeObject.palindromeOrNot(inputNumber);
+        if (finalNumber == temporaryNumber)
+            System.out.println("Palindrome number ");
+        else
+            System.out.println("Not a Palindrome number ");
+    }
+
+    int sum = 0, remainder;
```
Comment (prashantkawa) [inline checkPalindrome/CheckPalindromeProgram.java]: use private here

File: checkPalindrome/CheckPalindromeProgram.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    int palindromeOrNot(int inputNumber) {
```
Comment (prashantkawa) [inline checkPalindrome/CheckPalindromeProgram.java]: public is missing

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: checkPalindrome/CheckPalindromeProgram.java (hunk 1 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: used recursion.

File: checkPalindrome/CheckPalindromeProgram.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+   private int sum = 0, remainder;
+
+   public int palindromeOrNot(int inputNumber) {
+        if (inputNumber != 0) {
+            remainder = inputNumber % 10;
+            sum = (sum * 10) + remainder;
+            inputNumber /= 10;
+            palindromeOrNot(inputNumber);
+        }
+        return sum;
```
Comment (prashantkawa) [inline checkPalindrome/CheckPalindromeProgram.java]: Please try with negative number

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: checkPalindrome/CheckPalindromeProgram.java (hunk 2 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: used private

File: checkPalindrome/CheckPalindromeProgram.java (hunk 3 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: used public

File: checkPalindrome/CheckPalindromeProgram.java (hunk 4 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: Added logic for negative number

File: checkPalindrome/CheckPalindromeProgram.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+   public int palindromeOrNot(int inputNumber) {
+
+       if (inputNumber != 0) {
+            remainder = inputNumber % 10;
+            sum = (sum * 10) + remainder;
+            inputNumber /= 10;
+            palindromeOrNot(inputNumber);
+        }
+        return sum;
```
Comment (amitsahu8) [inline checkPalindrome/CheckPalindromeProgram.java:33]: Explain the logic written inside if

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: checkPalindrome/CheckPalindromeProgram.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package checkPalindrome;
+
```
Comment (jadhavrajashree40) [inline checkPalindrome/CheckPalindromeProgram.java:2]: Add Problem statement

Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java:2]: Added

File: checkPalindrome/CheckPalindromeProgram.java (hunk 5 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java:33]: if (inputNumber != 0) {
            remainder = inputNumber % 10;                      //remainder=1,2,1
            sum = (sum * 10) + remainder;                       //sum=1,12,121
            inputNumber /= 10;                                       //inputnumber=12,1
            palindromeOrNot(inputNumber);
        }
        return sum;

Till number becomes 0 if statement executes and we got sum exactly equals to inputNumber so its palindrome number. I written values of remainder, sum inputnumber in comment for each iteration followed by comma

**Review by jadhavrajashree40 — APPROVED** [review]

File: checkPalindrome/CheckPalindromeProgram.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        if (finalNumber == temporaryNumber)
+            System.out.println("Palindrome number ");
+        else
+            System.out.println("Not a Palindrome number ");
+    }
+
+    private int sum = 0, remainder;
+
+    public int palindromeOrNot(int inputNumber) {
+
```
Comment (vrushalee123) [inline checkPalindrome/CheckPalindromeProgram.java]: Why extra space. Please do proper indentation.

File: checkPalindrome/CheckPalindromeProgram.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package checkPalindrome;
+
+/**
+ * problem statement:
+ * Java program to check if the number is palindrome or not.
+ */
+public class CheckPalindromeProgram {
+    public static void main(String[] args) {
+        int inputNumber = 121, temporaryNumber, finalNumber;
```
Comment (vrushalee123) [inline checkPalindrome/CheckPalindromeProgram.java]: declare variable as private

**Review by vrushalee123 — CHANGES_REQUESTED** [review]

File: checkPalindrome/CheckPalindromeProgram.java (hunk 8 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: declared as a private

File: checkPalindrome/CheckPalindromeProgram.java (hunk 7 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java]: removed

File: checkPalindrome/CheckPalindromeProgram.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    private int sum = 0, remainder;
+
+    public int palindromeOrNot(int inputNumber) {
+        if (inputNumber != 0) {
+            remainder = inputNumber % 10;  //rem=1,2,1
+            sum = (sum * 10) + remainder;   //sum=1,12,121
+            inputNumber /= 10;              //inputnumber=12,1
+            palindromeOrNot(inputNumber);
+        }
+        return sum;
```
Comment (vrushalee123) [inline checkPalindrome/CheckPalindromeProgram.java:33]: Why cant see closing bracket

**Review by vrushalee123 — APPROVED** [review]

File: checkPalindrome/CheckPalindromeProgram.java (hunk 9 above)
Comment (tejashrighogare) [reply checkPalindrome/CheckPalindromeProgram.java:33]: In the .java file line no 34 and 35 is there for closing curly braces.

**Pull request merged**